core/domain/feedback_services.py

```diff
@@ -1004,14 +1013,22 @@ def _get_all_recipient_ids(exploration_id, thread_id, author_id):
                 in this thread, excluding owners of the exploration and the
                 given author.
     """
-    exploration_rights = rights_manager.get_exploration_rights(exploration_id)
-
-    owner_ids = set(exploration_rights.owner_ids)
     participant_ids = {
         message.author_id for message in get_messages(thread_id)
         if user_services.is_user_registered(message.author_id)
     }
 
+    suggestion_model = suggestion_models.GeneralSuggestionModel.get_by_id(
+        thread_id)
+    owner_ids = set([])
+
+    if suggestion_model is None or (
```

Comment:
Please add a comment explaining the logic here (e.g. why are these checks more correct than the previous implementation?). It seems subtle, and I want to make sure we capture this logic as others come across this code in the future.